src/DataLib/fendl32B_retrofit/tendl_preprocessing.py

```diff
@@ -0,0 +1,319 @@
+# Import packages
+import csv
+import aiohttp
+import asyncio
+import urllib.error
+import urllib.request
+import sys
+from logging_config import logger, LoggerWriter
+import ENDFtk
+import contextlib
+import os
+
+# Define constant(s)
+TENDL_GEN_URL = 'https://tendl.web.psi.ch/tendl_2017/neutron_file/'
+
+def load_csv(csv_path):
+    """
+    Load in the mt_table.csv file and store it in a dictionary.
+    
+    Arguments:
+        csv_path (str): File path to mt_table.csv
+            This should be in the same repository.
+    
+    Returns:
+        mt_dict (dict): Dictionary formatted data structure for mt_table.csv
+    """
+    mt_dict = {}
+
+    with open(csv_path, 'r') as f:
+        csv_reader = csv.DictReader(f)
+        for row in csv_reader:
+            mt_dict[row['MT']] = row['Reaction']
```

Comment:
process all the emitted particles here for once and for all, and store in a data structure that is something like:

you could actually store a dictionary:
```mt_data[MT] = { 'delKZA' : (change_P * 1000 + change_P + change_N) * 10 + change_M, 'emitted_particles' : emitted_particles }```